src/listener.py

```diff
@@ -0,0 +1,190 @@
+import select
+import socket
+from pascy.l2 import *
+import sys
+
+MY_ADDRESSES = {
+	"net1": ["02:42:37:94:2b:e4", "1.1.1.1"],
+	"net2": ["02:42:04:89:4c:a0", "2.2.2.1"]
+	}
+
+ROUTING_TABLE = {
+	"net1": 
+			{ "1.1.1.2": "02:42:01:01:01:02" 
+			},
+	"net2":
+			{ "2.2.2.2": "02:42:02:02:02:02"
+			}
+	}
+
+PACKETS_TO_SEND = []
+
+def get_network_by_ip(ip):
+	for network in ROUTING_TABLE:
+		if ip in ROUTING_TABLE[network].keys():
```

Comment:
This works well if your network has a subnet mask of 24 bits.
What happens if it 23? or something else?
Read about `ipaddress`.